### PR TITLE
Move packaging to framework level

### PR DESCRIPTION
**Pull request by HyperBrain**
## What did you implement:

Restores compatibility with provider plugins. See #3344 

## How did you implement it:

The packaging is now moved to the framework level. The global package plugin overwrites the `package:createDeploymentArtifacts`.

## How can we verify it:

Use "serverless deploy" with different plugins (including AWS). The command should work as before.
Additionally check if deploy and package still work for the AWS plugin with a given package option.

## Todos:

- [x] Write tests
- [ ] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

**Comment by HyperBrain** [comment]
@pmuens Could you do a check if this PR still works with the AWS deploy?

**Comment by pmuens** [comment]
@HyperBrain Just did a quick deploy to AWS. Everything fine for the AWS plugin.

Also biiiiig +1 for moving it to the Framework level since packaging is not AWS related. This change was also on my todo list, so happy to see that you're already working on it here 👍 .

**Comment by jthomas** [comment]
Quick test with the OpenWhisk Node.js template and... it works again!

I'll do a more comprehensive test tomorrow to ensure everything else is working.

**Comment by pmuens** [comment]
Just did a quick test with the Google Cloud Functions plugin. Works great. Thanks for the fix @HyperBrain 👍

**Comment by HyperBrain** [comment]
@jthomas @pmuens @eahefnawy From my side the PR is finalized now. Feel free to merge after you did the remaining tests (especially with the provider plugins).

File: lib/plugins/package/lib/packageService.js [hunk 1]
```diff
@@ -52,9 +52,12 @@ module.exports = {
   packageAll() {
     const exclude = this.getExcludes();
     const include = this.getIncludes();
-    const zipFileName = this.provider.naming.getServiceArtifactName();
+    const zipFileName = `${this.serverless.service.service}.zip`;
 
-    return this.zipDirectory(exclude, include, zipFileName);
+    return this.zipDirectory(exclude, include, zipFileName).then(filePath => {
```
Comment (dougmoscrop) [inline lib/plugins/package/lib/packageService.js]: if your promises are bluebird, you can use `.tap`

File: lib/plugins/package/lib/packageService.js [hunk 2]
```diff
@@ -52,9 +52,12 @@ module.exports = {
   packageAll() {
     const exclude = this.getExcludes();
     const include = this.getIncludes();
-    const zipFileName = this.provider.naming.getServiceArtifactName();
+    const zipFileName = `${this.serverless.service.service}.zip`;
```
Comment (pmuens) [inline lib/plugins/package/lib/packageService.js]: Maybe overkill but should we move this into a `provider.naming` function as well?

File: lib/plugins/package/lib/packageService.js [hunk 3]
```diff
@@ -63,7 +66,11 @@ module.exports = {
 
     const exclude = this.getExcludes(funcPackageConfig.exclude);
     const include = this.getIncludes(funcPackageConfig.include);
-    const zipFileName = this.provider.naming.getFunctionArtifactName(functionName);
-    return this.zipDirectory(exclude, include, zipFileName);
+    const zipFileName = `${functionName}.zip`;
```
Comment (pmuens) [inline lib/plugins/package/lib/packageService.js]: Same here. Overkill, but maybe a `provider.naming` is a good place to store this value.

File: lib/plugins/package/lib/packageService.js (hunk 2 above)
Comment (HyperBrain) [inline lib/plugins/package/lib/packageService.js]: This was in the naming namespace. But the problem is, that the topmost package command does not have knowledge nor access to the current provider. That's why we changed it here.
A possible improvement would be to create a provider-independent naming module that can be accessed here - but that will only increase debt as long as only one value/function is stored in there.

Comment (pmuens) [inline lib/plugins/package/lib/packageService.js]: Thanks for the explanation. Yes, I agree. A provider independent naming module would increase tech-debt and might be a little bit overkill for now. But something we should keep in mind if we see other patterns where we could utilize this again.

**Comment by HyperBrain** [comment]
@jthomas Could you comment here as soon as you did your final tests?

**Comment by eahefnawy** [comment]
Gonna merge this since it fixes the issue. Any further issues reported by @jthomas would be fixed in later PRs (if any).

@jthomas please let us know whenever you have updates 🙌 

As always, thanks a lot @HyperBrain 😊 ☺️

**Comment by jthomas** [comment]
No more issues encountered. 🥇
LGTM.

**Pull request merged**